spec/system/custom_sidebar_sections_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+describe "Custom sidebar sections", type: :system, js: true do
+  fab!(:user) { Fabricate(:user) }
+  let(:section_modal) { PageObjects::Modals::SidebarSectionForm.new }
+  let(:sidebar) { PageObjects::Pages::CustomSectionsSidebar.new }
+
+  before do
+    ### TODO remove when enable_custom_sidebar_sections SiteSetting is removed
+    group = Fabricate(:group)
+    group_user = Fabricate(:group_user, group: group, user: user)
+    SiteSetting.enable_custom_sidebar_sections = group.id.to_s
+    sign_in user
+  end
+
+  it "allows the user to create custom section" do
+    visit("/latest?enable_sidebar=1")
+    sidebar.open_new_custom_section
+
+    expect(page).to have_css(".sidebar-section-form-modal")
+    expect(page).to have_button("Save", disabled: true)
+
+    section_modal.fill_name("My section")
+
+    section_modal.fill_link(
+      "Sidebar Tags",
+      "http://#{Capybara.current_session.server.host}:#{Capybara.current_session.server.port}/tags",
+    )
+    expect(page).to have_button("Save", disabled: false)
+
+    section_modal.save
+
+    expect(page).to have_button("My section")
+    expect(page).to have_link("Sidebar Tags")
+  end
+
+  it "allows the user to delete custom section" do
+    visit("/latest?enable_sidebar=1")
+    sidebar.open_new_custom_section
+
+    section_modal.fill_name("My section")
+    section_modal.fill_link(
+      "Sidebar Tags",
+      "http://#{Capybara.current_session.server.host}:#{Capybara.current_session.server.port}/tags",
+    )
+    section_modal.save
```

Comment:
Instead of creating these records again via the system tests, we can fabricate them upfront making the test less expensive to run.